### PR TITLE
Add logic to keep opened screen after close and reopen plugin pane

### PR DESCRIPTION
**Pull request by vkatz**
- Keep ui state after hide/unhide pane
- OutputFormat key parse crush fix

https://github.com/user-attachments/assets/e6513228-6d50-419e-b4bf-b6276785e4eb

### CODE REVIEW

File: components/generator/imagevector/src/main/kotlin/io/github/composegears/valkyrie/generator/imagevector/ImageVectorGenerator.kt [hunk 1]
```diff
@@ -17,9 +17,7 @@ enum class OutputFormat(val key: String) {
     ;
 
     companion object {
-        fun from(key: String?): OutputFormat {
-            return OutputFormat.valueOf(key ?: BackingProperty.key)
-        }
+        fun from(key: String?) = OutputFormat.entries.find { it.key == key } ?: BackingProperty
```
Comment (Goooler) [inline components/generator/imagevector/src/main/kotlin/io/github/composegears/valkyrie/generator/imagevector/ImageVectorGenerator.kt:20]: Why need this update?

File: idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/settings/InMemorySettings.kt [hunk 2]
```diff
@@ -11,6 +11,8 @@ class InMemorySettings {
     private val _settings = MutableStateFlow(value = PersistentSettings.persistentSettings.toValkyriesSettings())
     val settings = _settings.asStateFlow()
 
+    private var uiState: Map<String, Any?> = emptyMap()
```
Comment (Goooler) [inline idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/settings/InMemorySettings.kt:14]:
```suggestion
    var uiState: Map<String, Any?> = emptyMap()
        private set
```

File: idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/settings/InMemorySettings.kt [hunk 3]
```diff
@@ -66,6 +68,12 @@ class InMemorySettings {
         _settings.updateState { PersistentSettings.persistentSettings.toValkyriesSettings() }
     }
 
+    fun getUIState() = uiState
+
```
Comment (Goooler) [inline idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/settings/InMemorySettings.kt:72]:
```suggestion
```

File: components/generator/imagevector/src/main/kotlin/io/github/composegears/valkyrie/generator/imagevector/ImageVectorGenerator.kt (hunk 1 above)
Comment (egorikftp) [reply components/generator/imagevector/src/main/kotlin/io/github/composegears/valkyrie/generator/imagevector/ImageVectorGenerator.kt:20]: @vkatz mentioned about runtime crash in this place, but I can't see the reason of it. Will check one more time

Comment (Goooler) [reply components/generator/imagevector/src/main/kotlin/io/github/composegears/valkyrie/generator/imagevector/ImageVectorGenerator.kt:20]: May cause the param isn't `null` but `""`.

Comment (egorikftp) [reply components/generator/imagevector/src/main/kotlin/io/github/composegears/valkyrie/generator/imagevector/ImageVectorGenerator.kt:20]: yes, looks like `var outputFormat: String? by string()` for new install return `""` instead `null`

File: idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/settings/InMemorySettings.kt (hunk 2 above)
Comment (egorikftp) [reply idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/settings/InMemorySettings.kt]: added

File: idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/settings/InMemorySettings.kt (hunk 3 above)
Comment (egorikftp) [reply idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/settings/InMemorySettings.kt]: added

**Comment by egorikftp** [comment]
Now plugin save navigation state after hide/unhide, but viewmodels clear data.
Will be covered with https://github.com/ComposeGears/Tiamat/issues/49

**Review by egorikftp — APPROVED** [review]

File: components/generator/imagevector/src/main/kotlin/io/github/composegears/valkyrie/generator/imagevector/ImageVectorGenerator.kt (hunk 1 above)
Comment (vkatz) [reply components/generator/imagevector/src/main/kotlin/io/github/composegears/valkyrie/generator/imagevector/ImageVectorGenerator.kt:20]: OutputFormat.valueOf takes 'enum name' and not a 'key' param, fun suplies exatly 'key' so it should be searched by 'key'

Comment (Goooler) [reply components/generator/imagevector/src/main/kotlin/io/github/composegears/valkyrie/generator/imagevector/ImageVectorGenerator.kt:20]: Make sense in this case.

**Review by Goooler — APPROVED** [review]

**Pull request merged**